src/Switch-Arduino_MCP4728/main.py

```diff
@@ -0,0 +1,269 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+
+# SweepMe! device class
+# Device: Arduino MCP4728
+
+
+from pysweepme import EmptyDevice
+
+
+class Device(EmptyDevice):
+    description = """
+        <h3>Arduino MCP4728</h3>
+        <p>This driver allows to set output voltages at MCP 4728 boards with 12-bit resolution. It can control up to 8 boards, each with 4 pins.</p>
+        <p>Set <em>Channel </em>to the pin number you want to set, or <em>all </em>to set all four channels. The voltage values must be passed as a colon-separated string: <em>1.0:2.5:0:4.2</em></p>
+        <p>The I&sup2;C address is set as integer 0-7, corresponding to the boards standard addresses 0x60-0x67 (HEX). You can check your devices' address by using an <a href="https://playground.arduino.cc/Main/I2cScanner/">I&sup2;C Scanner</a>.</p>
+        <p>The maximum voltage is defined by the Voltage reference, which can either be internal (2.048 V or 4.096 V by using 2x gain) or from an external source, e.g. the Arduino's 5 V or 3.3 V output. When using an external reference, the voltage must be given.</p>
+        <p>To use multiple MCPs, their I&sup2;C addresses need to be changed individually, as described <a href="https://github.com/jknipper/mcp4728_program_address">here</a>. Choose <em>Channel: all </em>and set a comma-separated list for <em>I2C Address: 0, 1, 2</em>.</p>
+    """
+
+    def __init__(self) -> None:
+        EmptyDevice.__init__(self)
+
+        self.shortname = "Arduino MCP4728"
+        self.variables = []
+        self.units = []
+
+        self.port_manager = True
+        self.port_types = ["COM"]
+        self.port_properties = {
+            "EOL": "\n",
+            "timeout": 15,
+            "baudrate": 115200,
+        }
+
+        self.multi_pins = False
+        self.multi_mcp = False
+        self.max_voltage = None
+        self.addresses = []
+
+        self.unit = {
+            "Voltage in V": "V",
+            "Output in %": "%",
+        }
+
+    def set_GUIparameter(self):
+        return {
+            "Channel": ["0", "1", "2", "3", "all"],
+            "I2C Address": "0",
+            "SweepMode": ["Voltage in V", "Output in %"],
+            "Voltage reference": ["External", "Internal 2.048 V", "Internal 4.096 V"],
```

Comment:
I suggest to put Internal 2.048 V at first place and External at the last place. If no external voltage is connected, it could otherwise float. So people that use External have to change it. Otherwise, 2.048 at max are applied